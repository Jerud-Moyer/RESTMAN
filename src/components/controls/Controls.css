.Control {
  margin: 20px;
}

.header {
  display: flex;
  justify-content: center;
  align-items: center;
  text-shadow: 2px 1px 2px blue;
}
.urlinput {
  width: 600px;
}
