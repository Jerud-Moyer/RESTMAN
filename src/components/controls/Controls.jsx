import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls =  ({
  url,
  onChange,
  onSubmit
}) => (
  <>
    <header>
      <h1 className={styles.header}>RESTMAN</h1>
    </header>
    <form className={styles.form} onSubmit={onSubmit}>
      <label htmlFor="url">URL</label>
      <input className={styles.urlinput}
        id="url"
        type="url"
        name="url"
        value={url}
        onChange={onChange}
      />
      <label className={styles.Control} htmlFor="method">GET</label>
      <input
        id="method"
        type="radio"
        name="method"
        value="GET"
        onChange={onChange}
      />
      <label className={styles.Control} htmlFor="method">POST</label>
      <input id="method"
        type="radio"
        name="method"
        value="POST"
        onChange={onChange}
      />
      <label className={styles.Control} htmlFor="method">PUT</label>
      <input id="method"
        type="radio"
        name="method"
        value="PUT"
        onChange={onChange}
      />
      <label className={styles.Control} htmlFor="method">DELETE</label>
      <input id="method"
        type="radio"
        name="method"
        value="DELETE"
        onChange={onChange}
      />
      <button type="submit">Submit</button>


    </form>
  </>
);

Controls.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Controls;
