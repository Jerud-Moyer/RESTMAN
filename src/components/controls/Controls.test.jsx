import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Controls from './Controls';

describe('Controls component', () => {
  afterEach(() => cleanup());

  it('renders Controls', () => {
    const { asFragment } = render(<Controls
      url="Test url"
      method="posty"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
