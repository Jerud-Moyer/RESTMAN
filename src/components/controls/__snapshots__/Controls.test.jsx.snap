// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="header"
    >
      RESTMAN
    </h1>
  </header>
  <form
    class="form"
  >
    <label
      for="url"
    >
      URL
    </label>
    <input
      class="urlinput"
      id="url"
      name="url"
      type="url"
      value="Test url"
    />
    <label
      class="Control"
      for="method"
    >
      GET
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="GET"
    />
    <label
      class="Control"
      for="method"
    >
      POST
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="POST"
    />
    <label
      class="Control"
      for="method"
    >
      PUT
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="PUT"
    />
    <label
      class="Control"
      for="method"
    >
      DELETE
    </label>
    <input
      id="method"
      name="method"
      type="radio"
      value="DELETE"
    />
    <button
      type="submit"
    >
      Submit
    </button>
    <textarea
      class="JsonInput"
      name="body"
      placeholder="raw JSON body"
    />
  </form>
</DocumentFragment>
`;
